CONTRIBUTING.md

```diff
@@ -0,0 +1,155 @@
+a separate branch and create a pull request.
+Please also use labels for your pull request, name issues that you refer to
+and assign reviewers who would be able to review your suggested test cases.
+Of course, before you do that we ask you to browse through the existing
+test cases first to make sure that the test cases you would like to propose
+are not covered by other test cases already.
+
+## Contributions Workflow via Pull Requests
+
+If you would like to contribute with pull requests for enhancements, bug-fixes
+or test cases, we kindly ask you to follow this workflow:
+
+![Development Workflow](docs/images/workflow.png)
+
+## Guidelines
+
+### Ansible Guides
+
+Because this is an Ansible project we agree with the 
+[Ansible Community Guide](https://docs.ansible.com/ansible/latest/community/index.html)
+and adhere to the
+[Ansible style guide](https://ansible.readthedocs.io/en/stable/dev_guide/style_guide/index.html).
+
+### Linting the Project
+
+Please make sure that your changes are linted before suggesting those changes
+in pull requests:
+
+```shell
+$ pipenv install --dev
+$ pipenv run molecule lint
+```
+
+### REUSE Specification Compliance
+
+Each file in this project needs to have a proper license and copyright header.
+Please check that each file you add to the project contains license and 
+copyright information and that it thereby meets the
+[REUSE Specification](https://reuse.software/spec/).
+
+```shell
+$ pipenv install --dev
+$ pipenv run reuse lint
+```
+
+### Testing the Ansible Role
+
+During development of your changes please use the existing automated test cases
+to verify that your changes do not break existing test cases:
+
+```shell
+$ pipenv install --dev
+$ pipenv run molecule test
+````
+
+Please consider using
+[Ansible Molecule](https://molecule.readthedocs.io/en/latest/)
+to write test cases for your suggested changes.
+Feel free to add those tests to the GitHub Actions CI Pipeline of this project.
+
+### GitHub Actions CI Pipeline
+
+Make sure that all GitHub Actions CI pipeline jobs pass in your fork before
+creating pull requests and that all GitHub Actions CI pipeline jobs pass in
+your pull request before assigning reviewers to your pull request.
+
+## How are Contributors Given Credit for Their Valuable Work?
+
+Please add yourself to the list of contributors in file 
+[README.md](README.md#contributors) 
+via a pull request if you made significant contributions to this role.
+Significant contributions are done by suggesting pull requests that fix
+bugs or add features or test cases to the project.
+Since all other contributions are welcome and may be significant as well 
+you can request to be added as a contributor which is then decided on a 
+case-by-case basis.
+
+## Ground Rules and Contributions We are Not Looking For
+
+While we welcome and appreciate all contributions we want people to be kind,
+respectful and mindful and won't support people who aren't.
```

Comment:
Maybe you can gather some more specific inspiration from the [Carpentries' Code of Conduct](https://docs.carpentries.org/topic_folders/policies/code-of-conduct.html#code-of-conduct-detailed-view)

Reply:
I added three links with codes of conducts. Please see new commit 36c82f0facedc6e59b104c379fb2fe21d4e9c119.